lib/publishing/nuget/publish.sh

```diff
@@ -72,7 +72,7 @@ log=$(mktemp -d)/log.txt
 found=false
 for NUGET_PACKAGE_PATH in $(find dotnet -name *.nupkg -not -iname *.symbols.nupkg); do
     found=true
-    if [ -n "${CODE_SIGNING_SECRET_ID:-}" ]; then
+    if [[ -n "${CODE_SIGNING_SECRET_ID:-}" && "${FOR_REAL:-}" == "true" ]]; then
```

Comment:
The self-signed certificate that is dynamically generated in the test is not usable for signing. Figured its ok for dry runs to skip this step.